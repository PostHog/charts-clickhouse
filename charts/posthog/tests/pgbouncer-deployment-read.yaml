suite: PostHog pgbouncer read deployment definition
templates:
  - templates/pgbouncer-read-deployment.yaml
  - templates/secrets.yaml

tests:
  - it: should be empty by default
    template: templates/pgbouncer-read-deployment.yaml # TODO: remove once secrets.yaml will be fixed/removed
    set:
      cloud: local # TODO: remove once secrets.yaml will be fixed/removed
    asserts:
      - hasDocuments:
          count: 0

  - it: should not be empty if enabled
    template: templates/pgbouncer-read-deployment.yaml # TODO: remove once secrets.yaml will be fixed/removed
    set:
      cloud: local # TODO: remove once secrets.yaml will be fixed/removed
      pgbouncerRead.enabled: true
    asserts:
      - hasDocuments:
          count: 1

  - it: should match snapshot data
    template: templates/pgbouncer-read-deployment.yaml # TODO: remove once secrets.yaml will be fixed/removed
    set:
      cloud: local # TODO: remove once secrets.yaml will be fixed/removed
      pgbouncerRead.enabled: true
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot:
          path: spec.template.spec

  - it: should be the correct kind
    template: templates/pgbouncer-read-deployment.yaml # TODO: remove once secrets.yaml will be fixed/removed
    set:
      cloud: local # TODO: remove once secrets.yaml will be fixed/removed
      pgbouncerRead.enabled: true
    asserts:
      - hasDocuments:
          count: 1
      - isKind:
          of: Deployment

  - it: should have a pod securityContext
    template: templates/pgbouncer-read-deployment.yaml # TODO: remove once secrets.yaml will be fixed/removed
    set:
      cloud: local # TODO: remove once secrets.yaml will be fixed/removed
      pgbouncerRead.enabled: true
      pgbouncerRead:
        enabled: true
        podSecurityContext: 
          enabled: true
          runAsUser: 1001
          fsGroup: 2000
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: spec.template.spec.securityContext.runAsUser
          value: 1001
      - equal:
          path: spec.template.spec.securityContext.fsGroup
          value: 2000

  - it: should have a container securityContext
    template: templates/pgbouncer-read-deployment.yaml # TODO: remove once secrets.yaml will be fixed/removed
    set:
      cloud: local # TODO: remove once secrets.yaml will be fixed/removed
      pgbouncerRead.enabled: true
      pgbouncerRead:
        enabled: true
        securityContext:
          enabled: true
          runAsUser: 1001
          allowPrivilegeEscalation: false
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: spec.template.spec.containers[0].securityContext.runAsUser
          value: 1001
      - equal:
          path: spec.template.spec.containers[0].securityContext.allowPrivilegeEscalation
          value: false

  - it: should not have a pod securityContext
    template: templates/pgbouncer-read-deployment.yaml # TODO: remove once secrets.yaml will be fixed/removed
    set:
      cloud: local # TODO: remove once secrets.yaml will be fixed/removed
      pgbouncerRead.enabled: true
      pgbouncerRead.podSecurityContext.enabled: false
    asserts:
      - hasDocuments:
          count: 1
      - isEmpty:
          path: spec.template.spec.securityContext
          value: 1001

  - it: should not have a container securityContext
    template: templates/pgbouncer-read-deployment.yaml # TODO: remove once secrets.yaml will be fixed/removed
    set:
      cloud: local # TODO: remove once secrets.yaml will be fixed/removed
      pgbouncerRead.enabled: true
      pgbouncerRead.securityContext.enabled: false
    asserts:
      - hasDocuments:
          count: 1
      - isEmpty:
          path: spec.template.spec.containers[0].securityContext

  - it: pgbouncer should have the correct resources if set
    template: templates/pgbouncer-read-deployment.yaml # TODO: remove once secrets.yaml will be fixed/removed
    set:
      cloud: local # TODO: remove once secrets.yaml will be fixed/removed
      pgbouncerRead.enabled: true
      pgbouncerRead.resources:
        requests:
          cpu: 100m
          memory: 50M
        limits:
          cpu: 512m
          memory: 100M
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: spec.template.spec.containers[0].resources
          value:
            requests:
              cpu: 100m
              memory: 50M
            limits:
              cpu: 512m
              memory: 100M

  - it: pgbouncer metrics exporter should have the correct resources if set
    template: templates/pgbouncer-read-deployment.yaml # TODO: remove once secrets.yaml will be fixed/removed
    set:
      cloud: local # TODO: remove once secrets.yaml will be fixed/removed
      pgbouncerRead.enabled: true
      pgbouncerRead.exporter:
        enabled: true
        resources:
          requests:
            cpu: 125m
            memory: 100M
          limits:
            cpu: 250m
            memory: 200M
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: spec.template.spec.containers[1].resources
          value:
            requests:
              cpu: 125m
              memory: 100M
            limits:
              cpu: 250m
              memory: 200M

  - it: allows setting pgbouncer exporter pullSecrets only
    template: templates/pgbouncer-read-deployment.yaml # TODO: remove once secrets.yaml will be fixed/removed
    set:
      cloud: local # TODO: remove once secrets.yaml will be fixed/removed
      pgbouncerRead:
        enabled: true
        exporter:
          enabled: true
          image:
            pullSecrets: [foo]
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: spec.template.spec.imagePullSecrets
          value: [name: foo]

  - it: allows setting pgbouncer pullSecrets only
    template: templates/pgbouncer-read-deployment.yaml # TODO: remove once secrets.yaml will be fixed/removed
    set:
      cloud: local # TODO: remove once secrets.yaml will be fixed/removed
      pgbouncerRead:
        enabled: true
        image:
          pullSecrets: [foo]
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: spec.template.spec.imagePullSecrets
          value: [name: foo]

  - it: allows setting pgbouncer and exporter secrets
    template: templates/pgbouncer-read-deployment.yaml # TODO: remove once secrets.yaml will be fixed/removed
    set:
      cloud: local # TODO: remove once secrets.yaml will be fixed/removed
      pgbouncerRead:
        enabled: true
        image:
          pullSecrets: [main]
        exporter:
          image:
            pullSecrets: [exporter]
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: spec.template.spec.imagePullSecrets
          value: 
          - name: main
          - name: exporter
