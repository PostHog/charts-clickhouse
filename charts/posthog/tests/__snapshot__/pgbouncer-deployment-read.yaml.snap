should match snapshot data:
  1: |
    containers:
    - env:
      - name: POSTGRESQL_USERNAME
        value: postgres
      - name: POSTGRESQL_PASSWORD
        valueFrom:
          secretKeyRef:
            key: postgresql-password
            name: RELEASE-NAME-posthog-postgresql
      - name: POSTGRESQL_DATABASE
        value: posthog
      - name: POSTGRESQL_HOST
        value: null
      - name: POSTGRESQL_PORT
        value: "5432"
      - name: PGBOUNCER_DATABASE
        value: posthog
      - name: PGBOUNCER_PORT
        value: "6543"
      - name: PGBOUNCER_MAX_CLIENT_CONN
        value: "1000"
      - name: PGBOUNCER_POOL_MODE
        value: transaction
      - name: PGBOUNCER_IGNORE_STARTUP_PARAMETERS
        value: extra_float_digits
      image: bitnami/pgbouncer:1.18.0
      imagePullPolicy: IfNotPresent
      lifecycle:
        preStop:
          exec:
            command:
            - sh
            - -c
            - sleep 30 && kill -INT 1 && sleep 31
      livenessProbe:
        failureThreshold: 3
        initialDelaySeconds: 60
        periodSeconds: 10
        successThreshold: 1
        tcpSocket:
          port: 6543
        timeoutSeconds: 2
      name: posthog-pgbouncer-read
      ports:
      - containerPort: 6543
        name: pgbouncer
      readinessProbe:
        failureThreshold: 3
        initialDelaySeconds: 10
        periodSeconds: 5
        successThreshold: 1
        tcpSocket:
          port: 6543
        timeoutSeconds: 2
    serviceAccountName: RELEASE-NAME-posthog
    terminationGracePeriodSeconds: 65
